### ISSUE TITLE
Feature: Add who executed the action on the onParticipantsChanged event

### ISSUE THREAD

**Issue by tcortega** [open-wa/wa-automate-nodejs#1271]
Title says it all, it'd be nice if we could see who executed the action, on both ParticipantsChanged event listeners... (global is the other one).

Since currently we can only see who suffered from the action, and what was the action.

**Comment by smashah** [comment]
@tcortega this used to be possible by hooking into the GP2 message, but since that no longer works it hasn't been possible to see who did the action of inviting, promoting, demoting, etc.

**Comment by tcortega** [comment]
The event still happens in the WhatsApp, isn't there anywhere you could listen it from? I thought of a way to do that but didn't test it yet, it'd be more like a work around.

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci minor`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [3.3.0](https://github.com/open-wa/wa-automate-nodejs/compare/3.2.6...3.3.0) (2021-02-04)

- build(deps-dev): bump release-it from 14.2.2 to 14.3.0 [`#1302`](https://github.com/open-wa/wa-automate-nodejs/pull/1302)
- build(deps-dev): bump @types/puppeteer from 5.4.2 to 5.4.3 [`#1300`](https://github.com/open-wa/wa-automate-nodejs/pull/1300)
- build(deps): bump socket.io from 3.1.0 to 3.1.1 [`#1301`](https://github.com/open-wa/wa-automate-nodejs/pull/1301)
- build(deps-dev): bump @types/node from 14.14.22 to 14.14.24 [`#1303`](https://github.com/open-wa/wa-automate-nodejs/pull/1303)
- build(deps): bump puppeteer from 6.0.0 to 7.0.0 [`#1305`](https://github.com/open-wa/wa-automate-nodejs/pull/1305)
- build(deps): bump puppeteer-extra-plugin-stealth from 2.7.3 to 2.7.4 [`#1306`](https://github.com/open-wa/wa-automate-nodejs/pull/1306)
- build(deps-dev): bump marked from 1.2.8 to 1.2.9 [`#1307`](https://github.com/open-wa/wa-automate-nodejs/pull/1307)
- Release 3.3.0 [`36034b3`](https://github.com/open-wa/wa-automate-nodejs/commit/36034b35a914926615941f976c72530c7668c2df)
- fix: (BREAKING) typo `(on)GlobalParicipantsChanged` fixed to `(on)GlobalParticipantsChanged` [`e5d5089`](https://github.com/open-wa/wa-automate-nodejs/commit/e5d508972ab9af05b9dd716ced9e35c5c4c8550d)
- fix: massive improvements to onGlobalParticipants `remove` and `add` now include `by` #1271 #1113 [`fe0b931`](https://github.com/open-wa/wa-automate-nodejs/commit/fe0b931cd223bc593274ee9707803d05d31ca29c)

**Comment by smashah** [comment]
@tcortega the callback now fires the `by` property which represents who exactly added or removed the participant (doesn't work on `promote`/`demote`)
